dts/bindings/adc/adi,ad4170-adc.yaml

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2025 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Bindings for the ADI AD4170 Analog-to-Digital Converter.
+
+  This device is controlled over SPI and exposes one or more ADC input channels.
+  Each child node corresponds to a channel and supports standard ADC properties
+  such as gain, reference, resolution, and input configuration.
+
+  Example binding:
+
+  &spi0 {
+      status = "okay";
+      #address-cells = <1>;
+      #size-cells = <0>;
+
+      ad4170: ad4170@0 {
+          compatible = "adi,ad4170-adc";
+          reg = <0x0>;
+          spi-max-frequency = <100000000>;
+          spi-cpol;
+          spi-cpha;
+          status = "okay";
+
+          #address-cells = <1>;
+          #size-cells = <0>;
+          #io-channel-cells = <1>;
+
+          channel@1 {
+              reg = <1>;
+              zephyr,gain = "ADC_GAIN_1";
+              zephyr,reference = "ADC_REF_INTERNAL";
+              zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+              zephyr,resolution = <24>;
+              zephyr,input-positive = <AD4170_ADC_AIN0>;
+              zephyr,input-negative = <AD4170_ADC_AIN1>;
+          };
+
+          channel@2 {
+              reg = <2>;
+              zephyr,gain = "ADC_GAIN_1";
+              zephyr,reference = "ADC_REF_INTERNAL";
+              zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+              zephyr,resolution = <24>;
+              zephyr,input-positive = <AD4170_ADC_AIN12>;
+              zephyr,input-negative = <AD4170_ADC_AIN13>;
+          };
+      };
+  };
+
+  For input constants such as `AD4170_ADC_AIN0`, refer to the binding header:
+  `include/zephyr/dt-bindings/adc/ad4170-adc.h`
+
+compatible: adi,ad4170-adc
+
+include: [adc-controller.yaml, spi-device.yaml]
+
+properties:
+  bipolar:
+    type: boolean
+    description: |
+      If true, the ADC operates in bipolar mode, otherwise it operates in
+      unipolar mode.
+
+  adc-mode:
+    type: int
+    enum: [0, 1, 2, 3, 4]
+    default: 0
```

Comment:
Need to explain in the description why the default value was selected. Please do this for all properties that have a default.
https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules